platforms/evm/src/signer.ts

```diff
@@ -34,10 +34,18 @@ export async function getEvmSigner(
   const chain = opts?.chain ?? (await EvmPlatform.chainFromRpc(rpc))[1];
   const managedSigner = new NonceManager(signer);
 
+  if (managedSigner.provider === null) {
```

Comment:
you want to explicitly check for `null` here? it'll never be `undefined`?

Reply:
suppose it could be but the apparent types returned are `Provider | null`